subprojects/core/src/main/java/org/gradle/api/internal/artifacts/JavaEcosystemSupport.java

```diff
@@ -118,6 +121,65 @@ private static void configureLibraryElements(AttributesSchema attributesSchema,
         });
     }
 
+    private static void configureView(AttributesSchema attributesSchema, final ObjectFactory objectFactory) {
+        AttributeMatchingStrategy<CompileView> viewSchema = attributesSchema.attribute(CompileView.VIEW_ATTRIBUTE);
+        viewSchema.getCompatibilityRules().add(CompileViewCompatibilityRules.class);
+        viewSchema.getDisambiguationRules().add(CompileViewDisambiguationRules.class, actionConfiguration -> {
+            actionConfiguration.params(objectFactory.named(CompileView.class, CompileView.JAVA_API));
+            actionConfiguration.params(objectFactory.named(CompileView.class, CompileView.JAVA_COMPLETE));
+        });
+    }
+
+    @VisibleForTesting
+    public static class CompileViewDisambiguationRules implements AttributeDisambiguationRule<CompileView>, ReusableAction {
```

Comment:
Wouldn't this be testable with the default access lvel?

Reply:
Updated these to package-private